workers/unity/Assets/Fps/Scripts/GameLogic/StateMachine/InitState.cs

```diff
@@ -0,0 +1,130 @@
+            }
+        }
+
+        public override void ExitState()
+        {
+            if (Owner.UseSessionBasedFlow)
+            {
+                sessionScreenController.browseButton.onClick.RemoveListener(BrowseDeployments);
+                sessionScreenController.quickJoinButton.onClick.RemoveListener(Connect);
+
+                deploymentDatas.Clear();
+            }
+        }
+
+        public override void Tick()
+        {
+            if (!Owner.UseSessionBasedFlow)
+            {
+                return;
+            }
+
+            if (deploymentDatas.Count == 0)
+            {
+                if (pit == null)
+                {
+                    pit = GetDevelopmentPlayerIdentityToken();
+                }
+                else if (loginTokens == null)
+                {
+                    loginTokens = GetDevelopmentLoginTokens(WorkerUtils.UnityClient, pit);
+                }
+                else
+                {
+                    FindDeployments();
+                    pit = null;
+                    loginTokens = null;
+                }
+            }
+            else if (!updatedDeploymentList)
+            {
+                sessionScreenController.ConnectionStatusUIController.ShowDeploymentListAvailableText();
+                sessionScreenController.browseButton.enabled = true;
+                sessionScreenController.quickJoinButton.enabled = true;
+
+                deploymentDatas.Sort((deployment1, deployment2) =>
+                    String.Compare(deployment1.Name, deployment2.Name, StringComparison.Ordinal));
+                Controller.FrontEndController.DeploymentListScreenController.SetDeployments(deploymentDatas.ToArray());
+
+                updatedDeploymentList = true;
+            }
+        }
+
+        private void BrowseDeployments()
+        {
+            Owner.SetState(new ListDeploymentsState(Controller, Owner));
+        }
+
+        private void Connect()
+        {
+            Owner.SetState(new ConnectState(null, Controller, Owner));
+        }
+
+        private void FindDeployments()
+        {
+            foreach (var loginToken in loginTokens)
+            {
+                var playerCount = 0;
+                var maxPlayerCount = 0;
+                var isAvailable = false;
+                foreach (var tag in loginToken.Tags)
+                {
+                    if (tag.StartsWith("players"))
+                    {
+                        playerCount = int.Parse(tag.Split('_').Last());
+                    }
+                    else if (tag.StartsWith("max_players"))
+                    {
+                        maxPlayerCount = int.Parse(tag.Split('_').Last());
+                    }
+                    else if (tag.StartsWith("status"))
+                    {
+                        var state = tag.Split('_').Last();
+                        isAvailable = state == "lobby" || state == "running";
```

Comment:
would prefer if we can enumerate the different state strings as variables somewhere